pandas/tests/indexing/test_indexing.py

```diff
@@ -1218,6 +1217,25 @@ def test_loc_getitem_bool(self):
         self.check_result('bool', 'loc', b, 'ix', b, typs=['empty'],
                           fails=KeyError)
 
+    def test_recursion_fails_loc(self):
+        # GH14554
+        # RuntimeError should bubble up if appropriate, rather
+        # than method specific exception
+        
+        recursive_df = pd.DataFrame({'a': [1,]}, index=['a'])
+        
+        def recursive_indexing_function(self, a):
+            return recursive_indexing_function(
+                recursive_df[['a'], ['a']])
+
+        oldVal=sys.getrecursionlimit()
```

Comment:
why are you changing the recursion limit at all?
just let the assert happen naturally.